kernel/src/kernel.rs

```diff
@@ -41,10 +46,43 @@ use crate::utilities::cells::NumericCellExt;
 /// is less than this threshold.
 pub(crate) const MIN_QUANTA_THRESHOLD_US: u32 = 500;
 
+#[derive(Copy, Clone)]
+pub struct StaticSlice<T> {
+    len: usize,
+    base: *mut T,
+}
+
+impl<T> StaticSlice<T> {
+    pub fn new<const N: usize>(slice: &'static mut [T; N]) -> Self {
+	StaticSlice {
+	    len: N,
+	    base: slice.as_mut_ptr(),
+	}
+    }
+
+    pub(crate) fn len(&self) -> usize {
+        self.len
+    }
+}
+
+impl<T> Deref for StaticSlice<T> {
+    type Target = [T];
+
+    fn deref(&self) -> &Self::Target {
+        unsafe { core::slice::from_raw_parts(self.base, self.len) }
+    }
+}
+
+impl<T> DerefMut for StaticSlice<T> {
+    fn deref_mut(&mut self) -> &mut Self::Target {
+        unsafe { core::slice::from_raw_parts_mut(self.base, self.len) }
+    }
+}
+
 /// Main object for the kernel. Each board will need to create one.
 pub struct Kernel {
     /// This holds a pointer to the static array of Process pointers.
-    processes: &'static [Option<&'static dyn process::Process>],
+    pub(crate) processes: &'static CoreLocal<MapCell<StaticSlice<Option<&'static dyn process::Process>>>>,
```

Comment:
I'm concerned about adding `MapCell` here. Nothing should ever call `.take()`, but that is an option now. What we need is a cell that can't lose its contents.